modules/xfem/src/geomsearch/XFEMElementPairLocator.C

```diff
@@ -41,5 +46,41 @@ XFEMElementPairLocator::reinit()
 
     ElementPairInfo new_elem_info(elem, q_points, weights, normal);
     _element_pair_info.insert(std::pair<const Elem*, ElementPairInfo>(elem, new_elem_info));
+
+    // elem2
+    _xfem->getXFEMIntersectionInfo(elem2, plane_id, normal, intersectionPoints, _use_displaced_mesh);
+
+    if (intersectionPoints.size() == 2)
+      _xfem->getXFEMqRuleOnLine(intersectionPoints, q_points, weights);
+    else if (intersectionPoints.size() > 2)
+      _xfem->getXFEMqRuleOnSurface(intersectionPoints, q_points, weights);
+
+    ElementPairInfo new_elem_info2(elem2, q_points, weights, normal);
+    _element_pair_info.insert(std::pair<const Elem*, ElementPairInfo>(elem2, new_elem_info2));
```

Comment:
My only major comment on this is that it seems a little odd that you're creating a second `ElementPairInfo` object for the second element. That's not really a problem, but I guess based on the naming of that object (it's for an element pair, after all), I'd expect to see a single instance of that object for each element/element constraint, and would expect that object to contain the information for both elements involved in the constraint. Following a pattern like that might make it easier to extend this to recursively split elements down the road as well.